elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -115,7 +147,15 @@ public void setSearchFactory(ExtendedSearchIntegrator boundSearchIntegrator) {
 		this.searchIntegrator = boundSearchIntegrator;
 		this.clientReference = searchIntegrator.getServiceManager().requestReference( JestClient.class );
 
-		initializeIndex();
+		// for dynamic sharding postInitialize() will not be invoked by the builder, so doing it here ourselves
+		if ( postInitialize != null ) {
+			try {
+				postInitialize.call();
+			}
+			catch (Exception e) {
+				throw new SearchException( "Index manager initialization failed", e );
```

Comment:
I've opened [HSEARCH-2168](https://hibernate.atlassian.net/browse/HSEARCH-2168) for log interface usage so we don't forget about it.